dotnetv3/Cognito/Tests/CognitoWrapperTests.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using Amazon.CognitoIdentityProvider;
+using Amazon.CognitoIdentityProvider.Model;
+using Microsoft.Extensions.Configuration;
+
+namespace SupportTests
+{
+    public class CognitoWrapperTests
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IAmazonCognitoIdentityProvider _client;
+        private readonly CognitoWrapper _wrapper;
+        private readonly string _userName;
+        private readonly string _password;
+        private readonly string _email;
+        private readonly string _clientId;
+        private readonly string _userPoolId;
+        private static string _mfaToken;
+        private static string _session;
+
+        /// <summary>
+        /// Constructor for the test class.
+        /// </summary>
+        public CognitoWrapperTests()
+        {
+            _configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("testsettings.json") // Load test settings from .json file.
+                .AddJsonFile("testsettings.local.json",
+                    true) // Optionally load local settings.
+                .Build();
+
+            _client = new AmazonCognitoIdentityProviderClient();
+            _wrapper = new CognitoWrapper(_client);
+
+            _userName = _configuration["UserName"];
+            _email = _configuration["Email"];
+            _password = _configuration["Password"];
+            _clientId = _configuration["ClientId"];
+            _userPoolId = _configuration["UserPoolId"];
+        }
+
+        [Fact()]
+        [Order(1)]
+        [Trait("Category", "Integration")]
+        public async Task SignUpAsyncTest()
+        {
+            var success = await _wrapper.SignUpAsync(_clientId, _userName, _password, _email);
+            Assert.True(success);
+        }
+
+        [Fact()]
+        [Order(2)]
+        [Trait("Category", "Integration")]
+        public async Task ListUserPoolsAsyncTest()
+        {
+            var userPools = await _wrapper.ListUserPoolsAsync();
+            Assert.NotNull(userPools);
+        }
+
+        [Fact()]
+        [Order(3)]
+        [Trait("Category", "Integration")]
+        public async Task GetAdminUserAsyncTest()
+        {
+            var userStatus = await _wrapper.GetAdminUserAsync(_userName, _userPoolId);
+            Assert.Equal(userStatus, UserStatusType.CONFIRMED);
+        }
+
+        [Fact()]
+        [Order(4)]
+        [Trait("Category", "Integration")]
+        public async Task ResendConfirmationCodeAsycTest()
+        {
+            var codeDeliveryDetails = await _wrapper.ResendConfirmationCodeAsyc(_clientId, _userName);
```

Comment:
Asyc -> Async

Reply:
Never noticed that one because it was mis-named in the CognitoWrapper class. Fixed all references.